### PR TITLE
Added Functionality to send email with Contact Form Details

### PR DESCRIPTION
**Pull request by Nishitbaria**
_Please_ read .env and add appropriate variables

# Description

In this pull request, I have added a new feature that allows the website's contact form details to be sent to the website's email address using Nodemailer and Gmail.
Fixes # (#510 )

### Changes Made:
- Created a new Node.js route for handling the contact form submission.
- Utilized the Nodemailer library to send emails.
- Configured the Gmail account as the email service provider.
- Secured sensitive information (Gmail credentials) using environment variables with the `dotenv` package.
- Implemented email validation and error handling for a smooth user experience.
- Styled the email content to make it visually appealing.

```
GMAIL_USER='enter your dummy gmail to send '
GMAIL_PASS='enter your password (you may need to genrate app specific password if normal password doesnt work)'
GMAIL_RECEIVER='enter email to recive contact form updates for eg:uba.gcoen@gmail.com'

```

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots

## Before
image

## After
image

###  Important points :
1.You can use any email for sender 
2.If normal password doesnt work ,generate app specific password    (read this article for reference:[article](https://nodemailer.com/usage/using-gmail/)
3.When you  merge my commit then in contactform.jsx, replace "https://localhost:5000/test/send-email" with "https://stichhub-backend.vercel.app/test/send-email"

### CODE REVIEW

**Comment by Nishitbaria** [comment]
@Naresh-chandanbatve,Please review and merge my pull request

File: server/routes/test.js [hunk 1]
```diff
@@ -1,8 +1,55 @@
...
+          <p style="font-family: Arial, sans-serif; font-size: 16px; font-weight: bold; color: #007bff;">Name:</p>
+          <p style="font-family: Verdana, sans-serif; font-size: 16px; color: #ff5733;">${name}</p>
+          <p style="font-family: Arial, sans-serif; font-size: 16px; font-weight: bold; color: #007bff;">Email:</p>
+          <p style="font-family: Verdana, sans-serif; font-size: 16px; color: #ff5733;">${email}</p>
+          <p style="font-family: Arial, sans-serif; font-size: 16px; font-weight: bold; color: #007bff;">Phone:</p>
+          <p style="font-family: Verdana, sans-serif; font-size: 16px; color: #ff5733;">${phoneno}</p>
+          <p style="font-family: Arial, sans-serif; font-size: 16px; font-weight: bold; color: #007bff;">Message:</p>
+          <p style="font-family: Verdana, sans-serif; font-size: 16px; color: #ff5733;">${message}</p>
+        </div>
+      </div>`
```
[the remark this answers is not in the document]
Comment (Nishitbaria) [reply server/routes/test.js:38]: I used HTML to format  email template ,There is no vulenerabilty

**Comment by Nishitbaria** [comment]
@Naresh-chandanbatve,Please review and merge my pull request

**Comment by Nishitbaria** [comment]
@Naresh-chandanbatve,Please review and merge my pull request

**Comment by Nishitbaria** [comment]
@Naresh-chandanbatve ,made sendEmail.js as it was,please review and merge my pull request

**Comment by Nishitbaria** [comment]
@Naresh-chandanbatve ,please review and merge my pull request

**Comment by Nishitbaria** [comment]
@Naresh-chandanbatve ,please review and merge my pull request

**Comment by Nishitbaria** [comment]
@Naresh-chandanbatve ,please review and merge my pull request

**Review by Naresh-chandanbatve — APPROVED** [review]

**Pull request merged**